molo/core/models.py

```diff
@@ -1736,6 +1736,15 @@ def get_parent_section(self, locale=None):
                     language__locale=locale).first()
             return self.get_parent().specific
 
+    def get_top_level_parent(self, locale=None, depth=3):
+        parent = self.get_parent().specific
```

Comment:
should this maybe be SectionPage.objects.filter(depth=depth).ancestor_of(self)?

That will make sure that you are always returning a section page as a parent and it should return the highest level section page?